### PR TITLE
search: add s-expression job printer

### PR DESCRIPTION
**Pull request by rvantonder**
## Test plan
Unit tests

### CODE REVIEW

File: internal/search/job/printers.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		case *subRepoPermsFilterJob:
+			b.WriteString("(FILTER")
+			depth++
+			writeSep(b, sep, indent, depth)
+			b.WriteString("SubRepoPermissions")
+			writeSep(b, sep, indent, depth)
+			writeSexp(j.child)
+			b.WriteString(")")
+			depth--
+		}
```
Comment (camdencheek) [inline internal/search/job/printers.go:120]: Default panicking case so we don't forget jobs?

File: internal/search/job/printers.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+// SexpFormat controls the s-expression format that represents a Job. `sep`
+// specifies the separator between terms. If `indent` is not empty, `indent` is
+// prefixed the number of times corresponding to depth of the term in the tree.
+// See the `Sexp` and `PrettySexp` convenience functions to see how these
+// options are used.
+func SexpFormat(job Job, sep, indent string) string {
+	b := new(bytes.Buffer)
+	depth := 0
+	var writeSexp func(Job)
```
Comment (camdencheek) [inline internal/search/job/printers.go:33]: Is this a "no recursive closures" thing?

**Review by camdencheek — APPROVED** [review]
coooooool 😎 

Why the all-caps for non-leaf jobs? Is this an s-expression convention?

File: internal/search/job/printers.go (hunk 2 above)
Comment (rvantonder) [reply internal/search/job/printers.go:33]: ayup 🙌 💯 🥇 🙇 ❤️

File: internal/search/job/printers.go (hunk 1 above)
Comment (rvantonder) [reply internal/search/job/printers.go:120]: yep sure

**Comment by rvantonder** [comment]
> Why the all-caps for non-leaf jobs? Is this an s-expression convention?

Nah. It's a grammar thing, sometimes there's a convention to capitalize nonterminals. I think it helps with readability but I'm not fussy about this. I will let @keegancsmith express preference too... Anyway since we have `autogold -update` this is easy to change :-)

**Comment by keegancsmith** [comment]
LGTM

**Pull request merged**